File: src/main/java/seremis/geninfusion/core/proxy/ClientProxy.java
package seremis.geninfusion.core.proxy;

import cpw.mods.fml.client.FMLClientHandler;
import cpw.mods.fml.client.registry.ClientRegistry;
import cpw.mods.fml.client.registry.RenderingRegistry;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.entity.Entity;
import seremis.geninfusion.client.render.RenderCrystal;
import seremis.geninfusion.client.render.RenderSoulCage;
import seremis.geninfusion.lib.RenderIds;
import seremis.geninfusion.soul.entity.EntitySoulCustom;
import seremis.geninfusion.soul.entity.EntitySoulCustomCreature;
import seremis.geninfusion.soul.entity.render.RenderEntitySoulCustom;
import seremis.geninfusion.tileentity.TileCrystal;
import seremis.geninfusion.tileentity.TileSoulCage;

public class ClientProxy extends CommonProxy {

    @Override
    @SideOnly(Side.CLIENT)
    public void registerRendering() {
        RenderingRegistry.registerBlockHandler(RenderIds.crystalRenderID, new RenderCrystal());
        RenderingRegistry.registerBlockHandler(RenderIds.soulCageRenderID, new RenderSoulCage());

        RenderingRegistry.registerEntityRenderingHandler(EntitySoulCustom.class, new RenderEntitySoulCustom());
        RenderingRegistry.registerEntityRenderingHandler(EntitySoulCustomCreature.class, new RenderEntitySoulCustom());

        ClientRegistry.bindTileEntitySpecialRenderer(TileCrystal.class, new RenderCrystal());
        ClientRegistry.bindTileEntitySpecialRenderer(TileSoulCage.class, new RenderSoulCage());
    }

    @Override
    public void removeEntity(Entity entity) {
        super.removeEntity(entity);

        if(isRenderWorld(entity.worldObj)) {
            entity.worldObj.removeEntity(entity);
        }
    }

    @Override
    public String playerName() {
        return FMLClientHandler.instance().getClient().thePlayer.getDisplayName();
    }

    @Override
    public void registerHandlers() {

    }
}

File: src/main/scala/seremis/geninfusion/core/proxy/ClientProxy.scala
package seremis.geninfusion.core.proxy

import net.minecraft.entity.Entity
import net.minecraft.item.Item
import net.minecraftforge.client.MinecraftForgeClient
import seremis.geninfusion.block.ModBlocks
import seremis.geninfusion.client.render.{RenderClayGolem, RenderCrystal}
import seremis.geninfusion.entity.EntityClayGolem
import seremis.geninfusion.lib.RenderIds
import seremis.geninfusion.soul.entity.render.RenderEntitySoulCustom
import seremis.geninfusion.soul.entity.{EntitySoulCustom, EntitySoulCustomCreature}
import seremis.geninfusion.tileentity.TileCrystal

class ClientProxy extends CommonProxy {

    @SideOnly(Side.CLIENT)
    override def registerRendering() {
        RenderingRegistry.registerBlockHandler(RenderIds.CrystalRenderID, new RenderCrystal)

        MinecraftForgeClient.registerItemRenderer(Item.getItemFromBlock(ModBlocks.crystal), new RenderCrystal)

        RenderingRegistry.registerEntityRenderingHandler(classOf[EntitySoulCustom], new RenderEntitySoulCustom)
        RenderingRegistry.registerEntityRenderingHandler(classOf[EntitySoulCustomCreature], new RenderEntitySoulCustom)

        RenderingRegistry.registerEntityRenderingHandler(classOf[EntityClayGolem], new RenderClayGolem)

        ClientRegistry.bindTileEntitySpecialRenderer(classOf[TileCrystal], new RenderCrystal)
    }

    override def removeEntity(entity: Entity) {
        super.removeEntity(entity)

        if(isRenderWorld(entity.worldObj)) {
            entity.worldObj.removeEntity(entity)
        }
    }

    override def playerName(): String = FMLClientHandler.instance().getClient.thePlayer.getDisplayName

    override def registerHandlers() {}
}
